commit a1d3afdb597e4034726e9c0923993d66f7e4a898 (refs/changes/26/4026/2)
Author: Jan Wyszynski <jan.wyszynski@joyent.com>
Date:   2018-05-30T19:30:07+00:00 (1 year, 4 months ago)
    
    joyent/pgstatsmon#13 pgstatsmon could poll pg_statio_user_tables and pg_statio_user_indexes

diff --git a/lib/queries.js b/lib/queries.js
index 077978e..c183692 100644
--- a/lib/queries.js
+++ b/lib/queries.js
@@ -82,6 +82,49 @@ var queries = [ {
 	{ 'attr': 'n_live_tup', 'help': 'estimated live tuples' },
 	{ 'attr': 'n_dead_tup', 'help': 'estimated dead tuples' }
     ]
+}, {
+    'name': 'pg_statio_user_tables',
+    'sql': [
+	'SELECT ',
+	'relname, ',
+	'heap_blks_read, ',
+	'heap_blks_hit, ',
+	'idx_blks_read, ',
+	'idx_blks_hit ',
+	'FROM pg_statio_user_tables'
+    ].join('\n'),
+    'statkey': 'relname',
+    'metadata': [ 'relname' ],
+    'counters': [
+	{ 'attr': 'heap_blks_read',
+	    'help': 'number of disk blocks read from this table' },
+	{ 'attr': 'heap_blks_hit',
+	    'help': 'number of buffer hits in this table' },
+	{ 'attr': 'idx_blks_read',
+	    'help': 'number of disk blocks read from all indexes on this ' +
+	    'table' },
+	{ 'attr': 'idx_blks_hit',
+	    'help': 'number of disk blocks hit in all indexes on this ' +
+	    'table' }
+    ]
+}, {
+    'name': 'pg_statio_user_indexes',
+    'sql': [
+	'SELECT ',
+	'indexrelname, ',
+	'relname, ',
+	'idx_blks_read, ',
+	'idx_blks_hit ',
+	'FROM pg_statio_user_indexes'
+    ].join('\n'),
+    'statkey': 'indexrelname',
+    'metadata': [ 'indexrelname', 'relname' ],
+    'counters': [
+	{ 'attr': 'idx_blks_read',
+	    'help': 'number of disk blocks read from this index' },
+	{ 'attr': 'idx_blks_hit',
+	    'help': 'number of buffer hits in this index' }
+    ]
 }, {
     'name': 'pg_stat_replication',
     'statkey': 'application_name',
